src/Stratis.Bitcoin.Tests.Common/TestBase.cs

```diff
@@ -25,33 +25,58 @@ public TestBase(Network network)
 
         public static string AssureEmptyDir(string dir)
         {
-            int deleteAttempts = 0;
-            while (deleteAttempts < 50)
+            var directoriesToDelete = GetDirectoriesToDelete(dir);
+
+            foreach (DirectoryInfo directoryWithTestNamePrefix in directoriesToDelete)
             {
-                if (Directory.Exists(dir))
+                int deleteAttempts = 0;
+                while (deleteAttempts < 50)
                 {
-                    try
+                    if (Directory.Exists(directoryWithTestNamePrefix.FullName))
                     {
-                        Directory.Delete(dir, true);
-                        break;
-                    }
-                    catch
-                    {
-                        deleteAttempts++;
-                        Thread.Sleep(200);
+                        try
+                        {
+                            Directory.Delete(directoryWithTestNamePrefix.FullName, true);
+                            break;
+                        }
+                        catch
+                        {
+                            deleteAttempts++;
+                            Thread.Sleep(200);
+                        }
                     }
+                    else
+                        break;
                 }
-                else
-                    break;
-            }
 
-            if (deleteAttempts >= 50)
-                throw new Exception(string.Format("The test folder: {0} could not be deleted.", dir));
+                if (deleteAttempts >= 50)
+                    throw new Exception(string.Format("The test folder: {0} could not be deleted.", directoryWithTestNamePrefix));
+
+            }
 
             Directory.CreateDirectory(dir);
             return dir;
         }
 
+        private static IEnumerable<DirectoryInfo> GetDirectoriesToDelete(string dir)
```

Comment:
Can you invert the if statement so you can return early and `Enumerable.Empty<DirectoryInfo>()` is not always created.

Something like this :

```
        private static IEnumerable<DirectoryInfo> GetDirectoriesToDelete(string dir)
        {
            var dirInfo = new DirectoryInfo(dir);
            var parentDirInfo = dirInfo.Parent;

            if (parentDirInfo == null && !parentDirInfo.Exists)
                return Enumerable.Empty<DirectoryInfo>();

            string guidSuffixPattern = "_????????-????-????-????-????????????";
            string searchPattern = dirInfo.Name.Substring(0, dirInfo.Name.LastIndexOf("_")) + guidSuffixPattern;

            return  parentDirInfo.EnumerateDirectories(searchPattern);
        }
```